ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,671 @@
+  {
+    // Set the terminal for the location. (For our case, use the "Load" terminal.)
+    if (utilityTerminal->name() == "Load")
+    {
+      loadTerminal = utilityTerminal;
+      break;
+    }
+  }
+
+
+  m_startingLocation = m_utilityNetwork->createElementWithAssetType(m_utilityAssetType, QUuid("{1CAF7740-0BF4-4113-8DB2-654E18800028}"), loadTerminal, this);
+
+  // Display starting location as graphic
+  m_utilityNetwork->featuresForElementsAsync(QList<UtilityElement*> {m_startingLocation}).then(this, [this](QList<ArcGISFeature*>)
+  {
+    ArcGISFeatureListModel* elementFeaturesList = m_utilityNetwork->featuresForElementsResult();
+    const Point startingLocationGeometry = geometry_cast<Point>(elementFeaturesList->first()->geometry());
+
+    GraphicsOverlay* overlay = new GraphicsOverlay(this);
+    SimpleMarkerSymbol* startingPointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Cross, Qt::green, 25, this);
+    Graphic* graphic = new Graphic(startingLocationGeometry, startingPointSymbol);
+    overlay->graphics()->append(graphic);
+
+    m_mapView->graphicsOverlays()->append(overlay);
+  });
+
+  UtilityNetworkDefinition* networkDefinition = m_utilityNetwork->definition();
+  UtilityDomainNetwork* domainNetwork = networkDefinition->domainNetwork("ElectricDistribution");
+  UtilityTier* tier = domainNetwork->tier("Medium Voltage Radial");
+
+  m_traceParameters = new UtilityTraceParameters(UtilityTraceType::Downstream, {m_startingLocation}, this);
+
+  m_traceConfiguration = tier->defaultTraceConfiguration();
+
+  m_traceParameters->setTraceConfiguration(m_traceConfiguration);
+
+  m_validateBtn = m_utilityNetwork->definition()->capabilities()->isSupportsValidateNetworkTopology();
+  emit validateBtn();
+  m_traceBtn = m_utilityNetwork->definition()->capabilities()->isSupportsTrace();
+  emit traceBtn();
+  m_stateBtn = m_utilityNetwork->definition()->capabilities()->isSupportsNetworkState();
+  emit stateBtn();
+  m_clearBtn = false;
+  emit clearBtn();
+
+  m_message = QString("Utility Network Loaded\n"
+      "Tap on a feature to edit.\n"
+      "Click 'Get State' to check if validating is\n"
+      "necessary or if tracing is available.\n"
+      "Click 'Trace' to run a trace.");
+  emit messageChanged();
+
+  m_busy = false;
+  emit isBusy();
+}
+
+LabelDefinition* ValidateUtilityNetworkTopology::createDeviceLabelDefinition()
+{
+  SimpleLabelExpression* labelExpression = new SimpleLabelExpression("[devicestatus]", this);
+
+  TextSymbol* textSymbol = new TextSymbol(this);
+  textSymbol->setSize(12);
+  textSymbol->setColor(Qt::blue);
+  textSymbol->setHaloColor(Qt::white);
+  textSymbol->setHaloWidth(2);
+
+  LabelDefinition* deviceLabelDefinition = new LabelDefinition(labelExpression, textSymbol, this);
+  deviceLabelDefinition->setUseCodedValues(true);
+  return deviceLabelDefinition;
+}
+
+LabelDefinition* ValidateUtilityNetworkTopology::createLineLabelDefinition()
```

Comment:
I think this function can be const. Everything that can be const, should be const.

Reply:
changed